packages/web3/src/address/address.test.ts

```diff
@@ -77,134 +150,237 @@ describe('address', function () {
-    expect(
-      isAssetAddress('2jW1n2icPtc55Cdm8TF9FjGH681cWthsaZW3gaUFekFZepJoeyY3ZbY7y5SCtAjyCjLL24c4L2Vnfv3KDdAypCddfAY')
-    ).toEqual(true)
-    expect(isAssetAddress('qeKk7r92Vn2Xjn4GcMEcJ2EwVfVs27kWUpptrWcWsUWC')).toEqual(true)
-    expect(isAssetAddress('vobthYg1e9tPKhmF96rpkv3akCj7vhvgPpsP4qwZqDw3')).toEqual(false)
-    expect(isContractAddress('15EM5rGtt7dPRZScE4Z9oL2EDfj84JnoSgq3NNgdcGFyu')).toEqual(false)
-    expect(
-      isContractAddress('2jW1n2icPtc55Cdm8TF9FjGH681cWthsaZW3gaUFekFZepJoeyY3ZbY7y5SCtAjyCjLL24c4L2Vnfv3KDdAypCddfAY')
-    ).toEqual(false)
-    expect(isContractAddress('qeKk7r92Vn2Xjn4GcMEcJ2EwVfVs27kWUpptrWcWsUWC')).toEqual(false)
-    expect(isContractAddress('vobthYg1e9tPKhmF96rpkv3akCj7vhvgPpsP4qwZqDw3')).toEqual(true)
-    expect(() => isAssetAddress('15EM5rGtt7dPRZScE4Z9oL2EDfj84JnoSgq3NNgdcGF')).toThrow('Invalid address:')
-    expect(() => isContractAddress('yya86C6UemCeLs5Ztwjcf2Mp2Kkt4mwzzRpBiG6qQ9k')).toThrow('Invalid address:')
+  describe('isValidAddress', () => {
+    it('should return true for valid addresses', () => {
+      expect(isValidAddress('15EM5rGtt7dPRZScE4Z9oL2EDfj84JnoSgq3NNgdcGFyu')).toBe(true)
+      expect(
+        isValidAddress('2jW1n2icPtc55Cdm8TF9FjGH681cWthsaZW3gaUFekFZepJoeyY3ZbY7y5SCtAjyCjLL24c4L2Vnfv3KDdAypCddfAY')
+      ).toBe(true)
+      expect(isValidAddress('eBrjfQNeyUCuxE4zpbfMZcbS3PuvbMJDQBCyk4HRHtX4')).toBe(true)
+      expect(isValidAddress('yya86C6UemCeLs5Ztwjcf2Mp2Kkt4mwzzRpBiG6qQ9kj')).toBe(true)
+    })
+
+    it('should return false for invalid addresses', () => {
+      expect(isValidAddress('')).toBe(false)
+      expect(isValidAddress('InvalidBase58!!')).toBe(false)
+      expect(isValidAddress('15EM5rGtt7dPRZScE4Z9oL2EDfj84JnoSgq3NNgdcGF')).toBe(false)
+      expect(isValidAddress('asdasdf')).toBe(false)
+    })
+  })
+
+  describe('isAssetAddress', () => {
+    it('should return true for valid asset addresses', () => {
+      expect(isAssetAddress('15EM5rGtt7dPRZScE4Z9oL2EDfj84JnoSgq3NNgdcGFyu')).toBe(true)
+      expect(
+        isAssetAddress('2jW1n2icPtc55Cdm8TF9FjGH681cWthsaZW3gaUFekFZepJoeyY3ZbY7y5SCtAjyCjLL24c4L2Vnfv3KDdAypCddfAY')
+      ).toBe(true)
+      expect(isAssetAddress('eBrjfQNeyUCuxE4zpbfMZcbS3PuvbMJDQBCyk4HRHtX4')).toBe(true)
+    })
+
+    it('should return false for contract addresses', () => {
+      expect(isAssetAddress('yya86C6UemCeLs5Ztwjcf2Mp2Kkt4mwzzRpBiG6qQ9kj')).toBe(false)
+    })
+
+    it('should throw error for invalid addresses', () => {
+      expect(() => isAssetAddress('InvalidBase58!!')).toThrow()
+    })
+  })
+
+  describe('isContractAddress', () => {
+    it('should return true for valid contract addresses', () => {
+      expect(isContractAddress('yya86C6UemCeLs5Ztwjcf2Mp2Kkt4mwzzRpBiG6qQ9kj')).toBe(true)
+    })
+
+    it('should return false for asset addresses', () => {
+      expect(isContractAddress('15EM5rGtt7dPRZScE4Z9oL2EDfj84JnoSgq3NNgdcGFyu')).toBe(false)
```

Comment:
Maybe we can test P2SH and P2MPKH address too just like we did for the `should return true for valid asset addresses` tests.

Reply:
We are already doing that in `isAssetAddress` tests